internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1024,30 +1025,57 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(ctx context.
 	return deployment, mutateFn, nil
 }
 
+func (r *AgentServiceConfigReconciler) getMustGatherImages(log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) string {
+	if instance.Spec.MustGatherImages == nil {
+		return MustGatherImages()
+	}
+	mustGatherVersions := make(versions.MustGatherVersions)
+	for _, specImage := range instance.Spec.MustGatherImages {
+		versionKey, err := getVersionKey(specImage.OpenshiftVersion)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("Problem parsing OpenShift version %v, skipping.", specImage.OpenshiftVersion))
+			continue
+		}
+		if mustGatherVersions[versionKey] == nil {
+			mustGatherVersions[versionKey] = make(versions.MustGatherVersion)
+		}
+		mustGatherVersions[versionKey][specImage.Name] = specImage.Url
+	}
+
+	if len(mustGatherVersions) == 0 {
+		return MustGatherImages()
+	}
+	encodedVersions, err := json.Marshal(mustGatherVersions)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Problem marshaling must gather images (%v) to string, returning default %v", mustGatherVersions, MustGatherImages()))
+		return OpenshiftVersions()
```

Comment:
Is this a typo? This should probably be `MustGatherImages()`, right?

Reply:
Good Catch. Fixing